api/v1alpha1/common_types.go

```diff
@@ -69,13 +71,21 @@ type FakeRepo struct {
 
 // HydratorMetadata contains metadata about the commit that is used to hydrate a branch. It is used to store
 type HydratorMetadata struct {
```

Comment:
Since this struct isn't used in any CRD, can we just move it to git.go? Lets us avoid a bunch of codegen, and makes it clear why it looks different from other stuff in this file (e.g. the Date field is a string instead of metav1.Time).